### PR TITLE
Push the collector docker image when sky-walking is tagged 

### LINKED ISSUE #162: Push collector docker image when sky-walking tagged
Auto push collector docker image when sky-walking tagged

### PR DESCRIPTION
**Pull request by ascrutae**
Add the following variables and the corresponding values into `Environment Variables` of corresponding project in Travis CI
```
DOCKER_PASSWORD: the password of docker user 
DOCKER_USERNAME: username of docker
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,3 +13,4 @@ script:
 
 after_success:
   - mvn clean cobertura:cobertura coveralls:report
+  - bash ./travis/deploy_image.sh
```
Comment (wu-sheng) [inline .travis.yml]: You are pushing image, not deploy image. Please choose the right word.

File: travis/deploy_image.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+check_pull_is_tag_or_release() {
+  if [ "${TRAVIS_TAG}" = "" ]; then
+    return 1
+  else
+    echo "[Deploying] deploy tag ${TRAVIS_TAG}."
```
Comment (wu-sheng) [inline travis/deploy_image.sh]: Same, `Deploy` and `Deploying`.

File: travis/deploy_image.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
+  mvn clean package docker:build
+  docker push skywalking/skywalking-collector:latest
+  docker push skywalking/skywalking-collector:${TRAVIS_TAG}
+}
+
+
+if check_pull_is_tag_or_release; then
+    deploy_collector_image
+    echo "[Deploying] deploy Done!"
```
Comment (wu-sheng) [inline travis/deploy_image.sh]: Same, `Deploy` and `Deploying`.

**Comment by wu-sheng** [comment]
Please provide document about how to config docker auth.

File: travis/push_image.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+check_pull_is_release() {
+  if [ "${TRAVIS_TAG}" = "" ]; then
+    return 1
+  else
+    echo "[Pushing] pushing docker image of ${TRAVIS_TAG}."
```
Comment (wu-sheng) [inline travis/push_image.sh]: `[Pushing] pushing` vs `[Pushing] push`(line 23), why?

File: travis/push_image.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    return 0
+  fi
+
+}
+
+push_collector_image() {
+  docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
Comment (wu-sheng) [inline travis/push_image.sh]: I assume we must config these two variables in travis-ci pages, right?

Comment (ascrutae) [inline travis/push_image.sh]: Please see the description of pull request.

**Pull request merged**